### PR TITLE
Add student facing script overview

### PR DESCRIPTION
**Pull request by dmcavoy**
Currently there is only one script overview that is used for both teachers and students. 

This PR:
- Creates a separate student facing overview for the script
- Copies all the existing script overviews (`description`) to also have an entry for student overview (`student_description`) in scripts.en.yml
- Creates the editing interface for new student overview and grouped overviews in their own editor section on the script overview page
- Made sure that saving works for new field
- Show the right overview based on if you are a student or teacher

### Screenshots of updates

Edit Page:
Screen Shot 2021-01-06 at 2 22 03 PM

Student View:
Screen Shot 2021-01-06 at 2 22 09 PM

Teacher View:
Screen Shot 2021-01-06 at 2 22 14 PM

### Updating scripts.en.yml
To copy the descriptions in scripts.en.yml I used a suggestion from @bethanyaconnor (Thanks!)
I updated `summarize_i18n_for_copy` to just be 2 lines:

```
data['description_student'] = data['description']
{'en' => {'data' => {'script' => {'name' => {new_name => data}}}}}
```

`summarize_i18n_for_copy` is called in `copy_and_write_i18n` to update the scripts.en.yml file. So in dashboard-console I ran:

```
Script.all.each do |script| script.copy_and_write_i18n(script.name) end
```

## Links

- [jira](https://codedotorg.atlassian.net/browse/PLAT-652)

## Testing story

- ScriptEditorTest updated
- ScriptOverviewHeaderTest updated

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

**Comment by davidsbailey** [comment]
Today, a teacher can see the teacher overview on CB and the student overview on code studio. with this change, the teacher will no longer be able to see the student overview anywhere, right? Do we think this is ok? CC @tess323 (apologies Tess, I can't remember if you receive github notifications)

**Comment by dmcavoy** [comment]
A teacher can see the student view of the script overview page by toggling into student view in the teacher panel. (This is already true for the lesson descriptions on the script overview page where there are different teacher and student descriptions for lessons) @davidsbailey @tess323 

Screen Shot 2021-01-06 at 11 04 23 PM

**Comment by dmcavoy** [comment]
Also today the overview on the script overview is shared between teacher and student. Not different for teachers and students.

**Comment by davidsbailey** [comment]
> A teacher can see the student view of the script overview page by toggling into student view in the teacher panel.

Good point! That addresses my concern.

> Also today the overview on the script overview is shared between teacher and student. Not different for teachers and students.

I agree with your statement. However, if you are suggesting that there is no teacher overview today, then I would disagree , because there is teacher-facing content for each unit on CB. Maybe now would be a good time to verify some assumptions. Is the following correct?
* the script overview at studio.code.org/s/coursef-2021 is (or will become) the "student overview" shown in this PR
* the teacher-facing content on https://curriculum.code.org/csd-20/unit1/ will become the "teacher overview" shown in this PR

**Comment by dmcavoy** [comment]
> I agree with your statement. However, if you are suggesting that there is no teacher overview today, then I would disagree , because there is teacher-facing content for each unit on CB. Maybe now would be a good time to verify some assumptions. Is the following correct?
> 
> * the script overview at studio.code.org/s/coursef-2021 is (or will become) the "student overview" shown in this PR
> * the teacher-facing content on https://curriculum.code.org/csd-20/unit1/ will become the "teacher overview" shown in this PR

This PR just duplicates the existing description today to be for both teachers and students. This leaves it up to curriculum writers how they want to use the fields going forward. This is also only the Unit page. The UnitGroup/Course page already has a separate teacher and student description which does not currently match what is on the Curriculum Builder page for teachers.

@davidsbailey

File: dashboard/config/locales/scripts.en.yml
Comment (davidsbailey) [inline dashboard/config/locales/scripts.en.yml:200]: what are the next steps here -- will `description` get updated with content from CB?

Comment (dmcavoy) [reply dashboard/config/locales/scripts.en.yml:200]: I was going to leave that up to curriculum writers to decide what they want to do once these fields are live. For now this does no harm in that it keeps the same description showing that was there before. As editors work on 2021 they can figure out if they just want the unit descriptions from CB moved over or not. Then if we figure out a general plan when we import the other past courses we can follow that plan

**Comment by davidsbailey** [comment]
> This PR just duplicates the existing description today to be for both teachers and students. This leaves it up to curriculum writers how they want to use the fields going forward. This is also only the Unit page. The UnitGroup/Course page already has a separate teacher and student description which does not currently match what is on the Curriculum Builder page for teachers.

OK, I think I see what you're saying. existing scripts will see no change, because the two descriptions will be the same. there are no guarantees what will happen to these fields for 2021 courses, since curriculum writers could change either or both fields however they want. this sounds ok to me.

File: dashboard/config/locales/scripts.en.yml
Comment (davidsbailey) [inline dashboard/config/locales/scripts.en.yml:209]: My understanding is that we'll want the added translations in this file to get merged in an earlier PR, then allow crowdin to populate translations, before merging the code changes in this PR. Otherwise we might have a period of time where student descriptions appeared in english when viewed in other languages. @bethanyaconnor or @Hamms is that correct?

**Review by davidsbailey — APPROVED** [review]
Thanks for jumping on this, Dani! LGTM after considering merging scripts.en.yml changes in an earlier PR.

**Comment by tess323** [comment]
Playing catchup -- Dani's approach seems correct to me based on our conversation at the bash.  Thank you both for being so thorough!

File: dashboard/config/locales/scripts.en.yml
Comment (bethanyaconnor) [reply dashboard/config/locales/scripts.en.yml:209]: That would be ideal but if it would block other work, I think it's okay to merge this whole change earlier. These strings will have translations once next week's sync runs as they're duplicates, so the turn around is going to be pretty small.

That said, if anything holds up this PR, it would be really great to check in this file before Sunday.

**Review by bethanyaconnor — APPROVED** [review]

**Comment by dmcavoy** [comment]
@davidsbailey As I mentioned earlier today I have just removed the work in ScriptOverviewHeader to show the student_description for students and instead it still shows the description for everyone. This PR https://github.com/code-dot-org/code-dot-org/pull/38460 has those changes reverted and can be the follow up once the i18n sync has fully happened

**Review by davidsbailey — APPROVED** [review]
looks great @dmcavoy !

**Pull request merged**
